modules/concurrency/src/io_ring/timer.cpp

```diff
@@ -147,13 +148,13 @@ auto Timer::tickSimulated(bool advance) -> bool {
   }
 
   if (advance) {
-    const auto& top = tasks_.front();
+    auto top = tasks_.front();
+    std::ranges::pop_heap(tasks_, std::greater<>{});
```

Comment:
interesting. do you need to protect pop_heap with std::ranges::is_heap or is that guaranteed?